test/e2e/build_dockerbx_test.go

```diff
@@ -0,0 +1,105 @@
+//go:build e2e
+
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"regexp"
+	"strings"
+	"testing"
+)
+
+func TestDockerBuildxBuildSuccessful(t *testing.T) {
+	env := BuildEnv(t)
+	kbld := Kbld{t, env.Namespace, env.KbldBinaryPath, Logger{}}
+
+	input := env.WithRegistries(`
+kind: Object
+spec:
+- image: simple-app-two
+- image: simple-app-three
+---
+apiVersion: kbld.k14s.io/v1alpha1
+kind: Sources
+sources:
+- image: simple-app-two
+  path: assets/simple-app
+  docker:
+    buildx:
+      file: dev/Dockerfile.dev
+- image: simple-app-three
+  path: assets/simple-app
+  docker:
+    buildx:
+      target: build-env
+`)
+
+	out, _ := kbld.RunWithOpts([]string{"-f", "-", "--images-annotation=false"}, RunOpts{
+		StdinReader: strings.NewReader(input),
+	})
+
+	out = regexp.MustCompile("sha256\\-[a-z0-9]{64}").ReplaceAllString(out, "SHA256-REPLACED")
+
+	expectedOut := `---
+kind: Object
+spec:
+- image: kbld:simple-app-two-SHA256-REPLACED
+- image: kbld:simple-app-three-SHA256-REPLACED
+`
+
+	if out != expectedOut {
+		t.Fatalf("Expected >>>%s<<< to match >>>%s<<<", out, expectedOut)
+	}
+}
+
+func TestDockerBuildxBuildAndPushSuccessful(t *testing.T) {
+	env := BuildEnv(t)
+	kbld := Kbld{t, env.Namespace, env.KbldBinaryPath, Logger{}}
+
+	input := env.WithRegistries(`
+kind: Object
+spec:
+- image: docker.io/*username*/kbld-e2e-tests-build
+- image: docker.io/*username*/kbld-e2e-tests-build2
+---
+apiVersion: kbld.k14s.io/v1alpha1
+kind: Sources
+sources:
+- image: docker.io/*username*/kbld-e2e-tests-build
+  path: assets/simple-app
+  docker:
+    buildx: {}
+- image: docker.io/*username*/kbld-e2e-tests-build2
+  path: assets/simple-app
+  docker:
+    buildx:
+      # try out multi platform build
+      rawOptions: ["--platform=linux/amd64,linux/arm64,linux/arm/v7"]
```

Comment:
It doesn't appear that this test makes any assertions that these platforms were in fact built.

Is this intentional / you don't want to test the underlying tool buildx itself?

I'm assuming you have manually tested passing rawOptions to buildx, and that it threads through correctly. (It looks like it should based off looking at the impl)

Reply:
> Is this intentional / you don't want to test the underlying tool buildx itself?

yeah, ive tried to avoid testing the tool functionality itself. just testing that it built something. any thoughts how far we should go?

Reply:
Yeah I ran into a bug with the [kubectl-buildkit](https://github.com/vmware-tanzu/buildkit-cli-for-kubectl) implementation in kbld that it wouldn't actually give you the digest for the multi-arch image, even if you passed those args correctly. I never looked into the bug. It seems to work correctly with this implementation, but it is a possibility apparently.